netlib-scalapack@2.2.0 %cce: libscalapack.so: undefined reference to `_idmin_'

### Steps to reproduce the issue

Having some trouble with `netlib-scalapack@2.2.0 %cce@14.0.0` on OLCF Crusher

Encountered similar errors in the past building `openblas%cce`:
* Related issue thread: https://github.com/xianyi/OpenBLAS/issues/3651
* PR that resolved issue for openblas: https://github.com/spack/spack/pull/31931

Concrete spec: [netlib-scalapack.spec.json.txt](https://github.com/spack/spack/files/9823998/netlib-scalapack.spec.json.txt)

Compiler definition for `%cce`
```
compilers:
  - compiler:
      spec: cce@14.0.0
      paths:
        cc: craycc
        cxx: crayCC
        f77: crayftn
        fc: crayftn
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-cray/8.3.3
      - cce/14.0.0
      - craype-x86-trento
      - libfabric
      - cray-pmi/6.1.2
      environment:
        set:
          RFE_811452_DISABLE: '1'
        append_path:
          LD_LIBRARY_PATH: /opt/cray/pe/gcc-libs:/opt/cray/libfabric/1.15.0.0/lib64
        prepend_path:
          LD_LIBRARY_PATH: /opt/cray/pe/pmi/6.1.2/lib
      extra_rpaths:
      - /opt/cray/pe/gcc-libs
      - /opt/cray/gcc-libs
```

<details><summary>Concretization</summary><pre>
 -   yueezqm  netlib-scalapack@2.2.0%cce@14.0.0~ipo~pic+shared build_type=Release patches=072b006,1c9ce5f arch=cray-sles15-zen3
[+]  lcohziv      ^cmake@3.23.3%cce@14.0.0~doc+ncurses+ownlibs~qt build_type=Release arch=cray-sles15-zen3
[+]  sp733kd          ^ncurses@6.3%cce@14.0.0~symlinks+termlib abi=none arch=cray-sles15-zen3
[+]  cph5uro              ^pkgconf@1.8.0%cce@14.0.0 arch=cray-sles15-zen3
[+]  xrx5l3d          ^openssl@1.1.1q%cce@14.0.0~docs~shared certs=mozilla patches=3fdcf2d arch=cray-sles15-zen3
[+]  ly34dvr              ^ca-certificates-mozilla@2022-07-19%cce@14.0.0 arch=cray-sles15-zen3
[+]  6ewdufh              ^perl@5.34.1%cce@14.0.0+cpanm+shared+threads arch=cray-sles15-zen3
[+]  eijh722                  ^berkeley-db@18.1.40%cce@14.0.0+cxx~docs+stl patches=b231fcc arch=cray-sles15-zen3
[+]  jx34xq2                  ^bzip2@1.0.8%cce@14.0.0~debug~pic+shared arch=cray-sles15-zen3
[+]  fxqvm5j                      ^diffutils@3.8%cce@14.0.0 arch=cray-sles15-zen3
[+]  a7qgfmg                          ^libiconv@1.16%cce@14.0.0 libs=shared,static arch=cray-sles15-zen3
[+]  76valko                  ^gdbm@1.19%cce@14.0.0 arch=cray-sles15-zen3
[+]  5n7wgk4                      ^readline@8.1.2%cce@14.0.0 arch=cray-sles15-zen3
[+]  dxqvezm                  ^zlib@1.2.12%cce@14.0.0+optimize+pic+shared patches=0d38234 arch=cray-sles15-zen3
[+]  ugxc7oe      ^cray-mpich@8.1.16%cce@14.0.0+wrappers arch=cray-sles15-zen3
[+]  ocnqdgc      ^openblas@0.3.20%cce@14.0.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared patches=f432226 symbol_suffix=none threads=openmp arch=cray-sles15-zen3
</pre></details>

Build error:
```
$> spack install -f ./netlib-scalapack.spec.json
...
==> Installing netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s
==> No binary for netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s found: installing from source
==> Using cached archive: /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/var/spack/cache/_source-cache/archive/40/40b9406c20735a9a3009d863318cb8d3e496fb073d201c5463df810e01ab2a57.tgz
==> Using cached archive: /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/var/spack/cache/_source-cache/archive/07/072b006e485f0ca4cba56096912a986e4d3da73aae51c2205928aa5eb842cefd
==> Applied patch /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/var/spack/repos/builtin/packages/netlib-scalapack/gcc10-compatibility.patch
==> Applied patch https://github.com/Reference-ScaLAPACK/scalapack/commit/d4d0066c041cf19a23f8b3aa62fbcf5f0a33c166.patch?full_index=1
==> netlib-scalapack: Executing phase: 'cmake'
==> netlib-scalapack: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

65 errors found in build log:
     4577    cd /tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm/TESTING/LIN && /gpfs/alpine/csc439/world-shared/E4S/ParaToo
             ls/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/cmake-3.23.3-lcohzivlorusntvn7pyb26dteqpk3qr2/bin/cmake -E cmake_link_script CMakeFiles/xcdtlu.dir/link.txt --v
             erbose=1
     4578    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/lib/spack/env/cce/ftn -dynamic CMakeFiles/xcgblu.dir/pcgbdriver.f.o CMakeFiles/xcgblu.dir/pcgbinfo.f.o CM
             akeFiles/xcgblu.dir/pcdblaschk.f.o CMakeFiles/xcgblu.dir/pcgbmv1.f.o CMakeFiles/xcgblu.dir/pcbmatgen.f.o CMakeFiles/xcgblu.dir/pcmatgen.f.o CMakeFiles/xcgblu.dir/pmatgeninc.f.o
              -o ../xcgblu  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm/lib:/gpfs/alpine/csc439/world-shared
             /E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2i7byrzq2dd/lib:/opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/l
             ib ../../lib/libscalapack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2
             i7byrzq2dd/lib/libopenblas.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb
             2i7byrzq2dd/lib/libopenblas.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpifort_cray.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpi_cray.so
     4579    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/lib/spack/env/cce/ftn -dynamic CMakeFiles/xcdtlu.dir/pcdtdriver.f.o CMakeFiles/xcdtlu.dir/pcdtinfo.f.o CM
             akeFiles/xcdtlu.dir/pcdtlaschk.f.o CMakeFiles/xcdtlu.dir/pcdbmv1.f.o CMakeFiles/xcdtlu.dir/pcbmatgen.f.o CMakeFiles/xcdtlu.dir/pcmatgen.f.o CMakeFiles/xcdtlu.dir/pmatgeninc.f.o
              -o ../xcdtlu  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm/lib:/gpfs/alpine/csc439/world-shared
             /E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2i7byrzq2dd/lib:/opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/l
             ib ../../lib/libscalapack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2
             i7byrzq2dd/lib/libopenblas.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb
             2i7byrzq2dd/lib/libopenblas.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpifort_cray.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpi_cray.so
     4580    [ 77%] Linking Fortran executable ../xclu
     4581    cd /tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm/TESTING/LIN && /gpfs/alpine/csc439/world-shared/E4S/ParaToo
             ls/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/cmake-3.23.3-lcohzivlorusntvn7pyb26dteqpk3qr2/bin/cmake -E cmake_link_script CMakeFiles/xclu.dir/link.txt --ver
             bose=1
     4582    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/lib/spack/env/cce/ftn -dynamic CMakeFiles/xclu.dir/pcludriver.f.o CMakeFiles/xclu.dir/pcluinfo.f.o CMakeF
             iles/xclu.dir/pcgetrrv.f.o CMakeFiles/xclu.dir/pcmatgen.f.o CMakeFiles/xclu.dir/pmatgeninc.f.o CMakeFiles/xclu.dir/pclaschk.f.o CMakeFiles/xclu.dir/pclafchk.f.o -o ../xclu  -Wl
             ,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm/lib:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.
             08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2i7byrzq2dd/lib:/opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib ../../lib/libsc
             alapack.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2i7byrzq2dd/lib/lib
             openblas.so /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-cray/spack/opt/spack/cray-sles15-zen3/cce-14.0.0/openblas-0.3.20-ocnqdgc3kg5mf6g3vyfqb2i7byrzq2dd/lib/li
             bopenblas.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpifort_cray.so /opt/cray/pe/mpich/8.1.16/ofi/crayclang/10.0/lib/libmpi_cray.so
  >> 4583    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4584    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4585    make[2]: *** [TESTING/LIN/CMakeFiles/xslu.dir/build.make:194: TESTING/xslu] Error 1
     4586    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4587    make[1]: *** [CMakeFiles/Makefile2:507: TESTING/LIN/CMakeFiles/xslu.dir/all] Error 2
     4588    make[1]: *** Waiting for unfinished jobs....
  >> 4589    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4590    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4591    make[2]: *** [TESTING/LIN/CMakeFiles/xdlu.dir/build.make:194: TESTING/xdlu] Error 1
     4592    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4593    make[1]: *** [CMakeFiles/Makefile2:533: TESTING/LIN/CMakeFiles/xdlu.dir/all] Error 2
  >> 4594    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4595    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4596    make[2]: *** [TESTING/LIN/CMakeFiles/xsdblu.dir/build.make:194: TESTING/xsdblu] Error 1
     4597    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4598    make[1]: *** [CMakeFiles/Makefile2:611: TESTING/LIN/CMakeFiles/xsdblu.dir/all] Error 2
  >> 4599    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4600    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4601    make[2]: *** [TESTING/LIN/CMakeFiles/xddtlu.dir/build.make:194: TESTING/xddtlu] Error 1
     4602    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4603    make[1]: *** [CMakeFiles/Makefile2:741: TESTING/LIN/CMakeFiles/xddtlu.dir/all] Error 2
  >> 4604    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4605    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4606    make[2]: *** [TESTING/LIN/CMakeFiles/xsdtlu.dir/build.make:194: TESTING/xsdtlu] Error 1
     4607    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4608    make[1]: *** [CMakeFiles/Makefile2:715: TESTING/LIN/CMakeFiles/xsdtlu.dir/all] Error 2
  >> 4609    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4610    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4611    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4612    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4613    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4614    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4615    make[2]: *** [TESTING/LIN/CMakeFiles/xddblu.dir/build.make:194: TESTING/xddblu] Error 1
     4616    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4617    make[1]: *** [CMakeFiles/Makefile2:637: TESTING/LIN/CMakeFiles/xddblu.dir/all] Error 2
  >> 4618    make[2]: *** [TESTING/LIN/CMakeFiles/xsgblu.dir/build.make:194: TESTING/xsgblu] Error 1
     4619    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4620    make[1]: *** [CMakeFiles/Makefile2:819: TESTING/LIN/CMakeFiles/xsgblu.dir/all] Error 2
  >> 4621    make[2]: *** [TESTING/LIN/CMakeFiles/xdgblu.dir/build.make:194: TESTING/xdgblu] Error 1
     4622    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4623    make[1]: *** [CMakeFiles/Makefile2:845: TESTING/LIN/CMakeFiles/xdgblu.dir/all] Error 2
  >> 4624    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4625    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4626    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4627    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4628    make[2]: *** [TESTING/LIN/CMakeFiles/xcdblu.dir/build.make:194: TESTING/xcdblu] Error 1
     4629    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4630    make[1]: *** [CMakeFiles/Makefile2:663: TESTING/LIN/CMakeFiles/xcdblu.dir/all] Error 2
  >> 4631    make[2]: *** [TESTING/LIN/CMakeFiles/xzdblu.dir/build.make:194: TESTING/xzdblu] Error 1
     4632    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4633    make[1]: *** [CMakeFiles/Makefile2:689: TESTING/LIN/CMakeFiles/xzdblu.dir/all] Error 2
  >> 4634    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4635    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4636    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4637    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4638    make[2]: *** [TESTING/LIN/CMakeFiles/xzlu.dir/build.make:194: TESTING/xzlu] Error 1
     4639    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4640    make[1]: *** [CMakeFiles/Makefile2:585: TESTING/LIN/CMakeFiles/xzlu.dir/all] Error 2
  >> 4641    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4642    make[2]: *** [TESTING/LIN/CMakeFiles/xzgblu.dir/build.make:194: TESTING/xzgblu] Error 1
     4643    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4644    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4645    make[1]: *** [CMakeFiles/Makefile2:897: TESTING/LIN/CMakeFiles/xzgblu.dir/all] Error 2
  >> 4646    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4647    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4648    make[2]: *** [TESTING/LIN/CMakeFiles/xzdtlu.dir/build.make:194: TESTING/xzdtlu] Error 1
     4649    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4650    make[1]: *** [CMakeFiles/Makefile2:793: TESTING/LIN/CMakeFiles/xzdtlu.dir/all] Error 2
  >> 4651    make[2]: *** [TESTING/LIN/CMakeFiles/xclu.dir/build.make:194: TESTING/xclu] Error 1
     4652    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4653    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4654    make[1]: *** [CMakeFiles/Makefile2:559: TESTING/LIN/CMakeFiles/xclu.dir/all] Error 2
  >> 4655    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4656    make[2]: *** [TESTING/LIN/CMakeFiles/xcgblu.dir/build.make:194: TESTING/xcgblu] Error 1
     4657    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4658    make[1]: *** [CMakeFiles/Makefile2:871: TESTING/LIN/CMakeFiles/xcgblu.dir/all] Error 2
  >> 4659    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_idmin_'
  >> 4660    /opt/cray/pe/cce/14.0.0/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: ../../lib/libscalapack.so: undefined reference to `_ismin_'
  >> 4661    make[2]: *** [TESTING/LIN/CMakeFiles/xcdtlu.dir/build.make:194: TESTING/xcdtlu] Error 1
     4662    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4663    make[1]: *** [CMakeFiles/Makefile2:767: TESTING/LIN/CMakeFiles/xcdtlu.dir/all] Error 2
     4664    make[1]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-netlib-scalapack-2.2.0-yueezqmhcs266p2k44he34tielmxys6s/spack-build-yueezqm'
  >> 4665    make: *** [Makefile:169: all] Error 2
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (c90fc445f8e297fdb7d1ebc6e97e91769a58ad00)
* **Python:** 3.9.7
* **Platform:** cray-sles15-zen3
* **Concretizer:** clingo

### Additional information

[CMakeOutput.log](https://github.com/spack/spack/files/9824010/CMakeOutput.log)
[spack-build-out.txt](https://github.com/spack/spack/files/9823983/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9823984/spack-build-env.txt)

@lukebroskop @wspear @mgates3

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/netlib-scalapack/package.py
+++ b/var/spack/repos/builtin/packages/netlib-scalapack/package.py
@@ -45,6 +45,13 @@ class ScalapackBase(CMakePackage):
         when="@2.2.0",
     )
 
+    def flag_handler(self, name, flags):
+        iflags = []
+        if name == "fflags":
+            if self.spec.satisfies("%cce"):
+                iflags.append("-hnopattern")
+        return (iflags, None, None)
+
     @property
     def libs(self):
         # Note that the default will be to search

